### PR TITLE
[dev] relocate interface info to core network

### PR DESCRIPTION
**Pull request by achilleasa**
## Description of change

This is a purely mechanical change that relocates `network.InterfaceInfo` (and friends like Route) to `core/network`. 

In addition, the PR also moves the InterfaceInfo -> params.NetworkConfig converter from `networkingcommon` into the `params` (params/network.go) package to avoid circular dependency issues with code in the `api/common` package that cannot import `networkingcommon` without causing an import cycle.

To avoid any confusion and maintain consistency with the other parts of the code-base I have intentionally imported `core/network` with the `corenetwork` alias even in files that don't import (or don't need to import after the change) `juju/network`.

### CODE REVIEW

**Comment by achilleasa** [comment]
While relocating the code, I noticed a comment that suggested renaming InterfaceInfo to InterfaceConfig for consistency purposes. If we want to do this, this PR would be a good place to start...

File: cloudconfig/cloudinit/cloudinit_centos.go [hunk 1]
```diff
@@ -8,12 +8,11 @@ import (
 	"fmt"
 	"strings"
 
+	corenetwork "github.com/juju/juju/core/network"
```
Comment (manadart) [inline cloudconfig/cloudinit/cloudinit_centos.go]: I'd just leave this unaliased.

File: cloudconfig/cloudinit/cloudinit_win.go [hunk 2]
```diff
@@ -5,9 +5,8 @@
 package cloudinit
 
 import (
+	corenetwork "github.com/juju/juju/core/network"
```
Comment (manadart) [inline cloudconfig/cloudinit/cloudinit_win.go]: Same.

File: cloudconfig/cloudinit/interface.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ package cloudinit
 
 import (
 	"github.com/juju/errors"
+	corenetwork "github.com/juju/juju/core/network"
```
Comment (manadart) [inline cloudconfig/cloudinit/interface.go]: Etc.

**Comment by achilleasa** [comment]
$$merge$$

**Pull request merged**